dashboard/test/models/user_helpers_test.rb

```diff
@@ -22,59 +22,65 @@ def create_user_with_username(username)
+    assert_equal 'grace',
+      UserHelpers.generate_username(User, 'Grace Hopper')
   end
 
   test 'generate_username for different usernames' do
     # Different usernames do not generate spurious collisions.
     create_user_with_username 'captain_picard'
     assert_equal 'captain', UserHelpers.generate_username(User, 'Captain')
-    assert_equal 'captain_p', UserHelpers.generate_username(User, 'Captain   P')
+    assert_equal 'grace', UserHelpers.generate_username(User, 'Grace')
   end
 
   test 'generate_username for existing username via dart throwing' do
-    create_user_with_username 'captain_picard'
+    create_user_with_username 'captain'
 
     # An existing username attempts the username, fails, and receives '784'.
     srand 0
-    assert_equal 'captain_picard784',
-      UserHelpers.generate_username(User, 'Captain Picard')
-    create_user_with_username 'captain_picard784'
+    assert_equal 'captain784',
+      UserHelpers.generate_username(User, 'Captain')
+    create_user_with_username 'captain784'
 
-    # The next Captain Picard attempts '784', fails, and receives '659'
+    # The next Captain attempts '784', fails, and receives '659'
     srand 0
-    assert_equal 'captain_picard659',
-      UserHelpers.generate_username(User, 'Captain Picard')
-    create_user_with_username 'captain_picard659'
+    assert_equal 'captain659',
+      UserHelpers.generate_username(User, 'Captain')
+    create_user_with_username 'captain659'
 
-    # The next Captain Picard attempts '784' and '659', fails, and
+    # The next Captain attempts '784' and '659', fails, and
     # receives '4264'.
     srand 0
-    assert_equal 'captain_picard4264',
-      UserHelpers.generate_username(User, 'Captain Picard')
-    create_user_with_username 'captain_picard4264'
+    assert_equal 'captain4264',
+      UserHelpers.generate_username(User, 'Captain')
+    create_user_with_username 'captain4264'
 
-    # The next Captain Picard attempts the above, fails and receives '5859'.
+    # The next Captain attempts the above, fails and receives '5859'.
     srand 0
-    assert_equal 'captain_picard5859',
-      UserHelpers.generate_username(User, 'Captain Picard')
+    assert_equal 'captain5859',
+      UserHelpers.generate_username(User, 'Captain')
   end
 
   test 'generate_username for existing username via fallback' do
-    ['', 784, 659, 4264, 5859, 51993, 96293, 54824, 47456, 383298, 593063, 548242, 474564].each do |suffix|
-      create_user_with_username "captain_picard#{suffix}"
+    ['', 784, 659, 4264, 5859, 51993, 96293, 54824, 47456, 383298, 593063, 548242].each do |suffix|
+      create_user_with_username "captain#{suffix}"
     end
 
     srand 0
-    assert_equal 'captain_picard383299',
-      UserHelpers.generate_username(User, 'Captain Picard')
+    assert_equal 'captain474564',
+      UserHelpers.generate_username(User, 'Captain')
   end
 
   test 'generate_username for long names' do
-    assert_equal 'this_is_a_really',
```

Comment:
The generated username is quite short here, do we have a minimum length for username? If the user's name is "A really long name", do we get "A" as username?

Reply:
If the generated name is less than 5 characters, then the name will be `coder_a`.